src/Stack/Package.hs

```diff
@@ -1108,42 +1141,45 @@ findCandidate dirs exts name = do
 -- | Warn the user that multiple candidates are available for an
 -- entry, but that we picked one anyway and continued.
 warnMultiple
-    :: MonadLogger m
+    :: (MonadLogger m, HasRunner env, MonadReader env m)
     => DotCabalDescriptor -> Path b t -> [Path b t] -> m ()
 warnMultiple name candidate rest =
-    $logWarn
-        ("There were multiple candidates for the Cabal entry \"" <>
-         showName name <>
-         "\" (" <>
-         T.intercalate "," (map (T.pack . toFilePath) rest) <>
-         "), picking " <>
-         T.pack (toFilePath candidate))
-  where showName (DotCabalModule name') = T.pack (D.display name')
-        showName (DotCabalMain fp) = T.pack fp
-        showName (DotCabalFile fp) = T.pack fp
-        showName (DotCabalCFile fp) = T.pack fp
+    -- TODO: figure out how to style 'name' and the dispOne stuff
+    $prettyWarnL
+        [ flow "There were multiple candidates for the Cabal entry \""
+        , fromString . showName $ name
+        , line <> bulletedList (map dispOne rest)
+        , line <> flow "picking:"
+        , dispOne candidate
+        ]
+  where showName (DotCabalModule name') = D.display name'
+        showName (DotCabalMain fp) = fp
+        showName (DotCabalFile fp) = fp
+        showName (DotCabalCFile fp) = fp
+        dispOne = fromString . toFilePath
```

Comment:
This is unfortunate, I can't figure out exactly why this can't be `display` instead of `dispOne` in this function. I get the error:

```
    • Couldn't match type ‘Ann (Path b t)’ with ‘AnsiAnn’
      Expected type: Path b t -> AnsiDoc
        Actual type: Path b t -> Doc (Ann (Path b t))
    • In the first argument of ‘map’, namely ‘display’
      In the first argument of ‘bulletedList’, namely
        ‘(map display rest)’
      In the second argument of ‘(<>)’, namely
        ‘bulletedList (map display rest)’
    • Relevant bindings include
        rest :: [Path b t] (bound at src/Stack/Package.hs:1146:29)
        candidate :: Path b t (bound at src/Stack/Package.hs:1146:19)
        warnMultiple :: DotCabalDescriptor
                        -> Path b t -> [Path b t] -> m ()
          (bound at src/Stack/Package.hs:1146:1)
```

when I try.

Reply:
Must be something about `Path b t` being polymorphic? A little confused.

Reply:
Yeah, it's because it's polymorphic.  I think adding `Display (Path b t)` to the function's constraints should solve the problem.  Alternatively, just make it monomorpic, I think it's just used for `(Path Abs File)` or something.

Reply:
Hmm, thanks for looking at it. I thought I tried that constraint and got the same error. Very possible I screwed it up though. I'll investigate more.